src/App.jsx

```diff
@@ -1,71 +1,102 @@
-        },
-        {
-          color: 'dark',
-          onClose: () => sessionStorage.setItem('newsletter-signup-dismissed', 'true')
+        let showNewsletterSignup = !(
+            newsletterSignupDismissed || newsletterSignupCompleted
+        );
+        if (show) {
+            showNewsletterSignup = true;
+        }
+        if (showNewsletterSignup) {
+            const modalRef = ModalService.open(
+                NewsletterSignupModal,
+                {
+                    onSuccessfulSignup: () => {
+                        closeNewsletterSignupModal(modalRef);
+                    },
+                },
+                {
+                    color: 'dark',
+                    onClose: () =>
+                        sessionStorage.setItem(
+                            'newsletter-signup-dismissed',
+                            'true'
+                        ),
+                }
+            );
         }
-      );
-    }
-  }
+    };
 
-  // Close newsletter sign-up modal after 5 seconds
-  const closeNewsletterSignupModal = (modalRef) => {
-    setTimeout(() => {
-      ModalService.close(modalRef);
-    }, 5000);
-  }
+    // Close newsletter sign-up modal after 5 seconds
+    const closeNewsletterSignupModal = (modalRef) => {
+        setTimeout(() => {
+            ModalService.close(modalRef);
+        }, 5000);
+    };
 
-  useEffect(() => {
-    // Display newsletter sign-up modal after 30 seconds
-    const timer = setTimeout(() => {
-      openNewsletterSignupModal();
-    }, 30000);
+    useEffect(() => {
+        // Display newsletter sign-up modal after 30 seconds
+        const timer = setTimeout(() => {
+            openNewsletterSignupModal();
+        }, 30000);
 
-    return () => {
-      clearTimeout(timer);
-    };
-  });
+        return () => {
+            clearTimeout(timer);
+        };
+    });
 
-  return (
-    <BrowserRouter>
-      <div className="App">
-        <AnnounceBar />
-        <TopNav />
-        <Switch>
-          <Route path="/" exact component={Homepage} />
-          <Route path="/campaigns" exact component={Campaigns} />
-          <Route path="/campaigns/:id" exact component={Campaign} />
-          <Route path="/aboutus" exact component={AboutUs} />
-          <Route path="/now-u-app" exact component={NowUApp} />
-          <Route path="/get-in-touch" exact component={GetInTouch} />
-          <Route path="/press" exact component={Press} />
-          <Route path="/faqs" exact component={FAQs} />
-          <Route path="/blog" exact component={Blog} />
-          <Route path="/blog/:id" exact component={BlogArticle} />
-          <Route exact component={err404} />
-        </Switch>
-      </div>
-      <Footer />
-    </BrowserRouter>
-  );
+    return (
+        <BrowserRouter>
+            <div className='App'>
+                <AnnounceBar />
+                <Navbar newsModal={openNewsletterSignupModal} />
```

Comment:
Thank you for showing me this and I feel like this is outside the scope of this PR, but should benefit us in the long run.
I am creating a utility functions file and going to export these modal functions. It will clean up the app file slightly as well as remove a lot of code and props from my components.

Reply:
I don't think we need a utils file. The functions are pretty small as they are and this functionality based on popping the modal are 30s is specific to this situation whereas your usage of the modal is specific for your situation at the moment. The code I suggested would just need to be put inside a function and be called `onClick` and it would be better than adding an override into the existing function which is a bit of a code smell.

Reply:
I bring this up because @stellenberger suggested I go with a DRY methodology. I have been trying to not copy code and had to rebuild this whole component system to stay DRY, at least I have been trying to keep it that way, I may have slipped some.